garmin_txt.cc

```diff
@@ -883,28 +880,16 @@ free_header(const header_type ht)
 
 /* data parsers */
 
-static int
-parse_date_and_time(char* str, time_t* value)
+static bool
+parse_date_and_time(char* str, QDateTime* value)
```

Comment:
I think it would be better to just return a possibly invalid QDateTime.  Then the callers just check isValid() instead of for nullptr, and use the return value if its Valid.

The first parameter can and should be const char*.